### PR TITLE
Bluetooth: controller: Add Direction Finding specific KConfig

### PR DESCRIPTION
**Pull request by ppryga-nordic**
This PR introduces separation of Direction Finding Kconfig entries from common Bluetooth controller Kconfig.
It should help to manage controllers Kconfig-s in future.

### CODE REVIEW

**Comment by ppryga** [comment]
Added HAS_HW_NRF_RADIO_BLE_DF Kconfig to nRF52833 Kconfig to improve scalability.

File: subsys/bluetooth/controller/Kconfig.ll_sw_split [hunk 1]
```diff
@@ -37,6 +37,15 @@ config BT_LLL_VENDOR_NORDIC
 	select BT_CTLR_XTAL_ADVANCED_SUPPORT
 	select BT_CTLR_SCHED_ADVANCED_SUPPORT
 	select BT_CTLR_TIFS_HW_SUPPORT
+	# Direction Finding
+	select BT_CTLR_DF_CTE_TX_SUPPORT if HAS_HW_NRF_RADIO_BLE_DF && \
```
Comment (kruithofa) [inline subsys/bluetooth/controller/Kconfig.ll_sw_split:41]: nothing selects HAS_HW_NRF_RADIO_BLE_DF, so these options do not become available

**Comment by joerchan** [comment]
Is this a duplicate of https://github.com/zephyrproject-rtos/zephyr/pull/29603?

**Comment by ppryga** [comment]
@joerchan 
> Is this a duplicate of #29603?
This is not duplicate. This PR was created before #29603. When I was waiting for review I have implemented next part of functionality and provided another PR. #29603 is based on the same commits that are in this PR so it looks like duplicate. 

In future I'll wait before posting consecutive PR-s until previous one is merged.

**Comment by carlescufi** [comment]
@ppryga could you please merge this PR with #29603, and introduce the Kconfig options in the commit where you are actually using them in the code?

**Comment by ppryga** [comment]
> @ppryga could you please merge this PR with #29603, and introduce the Kconfig options in the commit where you are actually using them in the code?

Would it be ok for you if I just close this PR, because its commits are also available in #29603? Or is it necessary to merge these commits into one that uses it?

**Comment by carlescufi** [comment]
> Would it be ok for you if I just close this PR, because its commits are also available in #29603? Or is it necessary to merge these commits into one that uses it?

Yes, you can just close it, no problem.

**Comment by ppryga** [comment]
Closing PR. Changes will be shipped by PR:  #29603

**Pull request closed without merging**